flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/data/DataStructureConvertersTest.java

```diff
@@ -355,7 +355,62 @@ static List<TestSpec> testData() {
+                                        DataTypes.FIELD("f5", DataTypes.FLOAT().notNull()),
+                                        DataTypes.FIELD("f6", DataTypes.DATE().notNull()),
+                                        DataTypes.FIELD("f7", DataTypes.BINARY(12).notNull()),
+                                        DataTypes.FIELD("f8", DataTypes.VARBINARY(12).notNull()),
+                                        DataTypes.FIELD("f9", DataTypes.CHAR(12).notNull()),
+                                        DataTypes.FIELD("f10", DataTypes.VARCHAR(12).notNull()),
+                                        DataTypes.FIELD("f11", DataTypes.BOOLEAN().notNull()),
+                                        DataTypes.FIELD("f12", DataTypes.TIME().notNull()),
+                                        DataTypes.FIELD("f13", DataTypes.TIMESTAMP().notNull()),
+                                        DataTypes.FIELD("f14", DataTypes.TIMESTAMP_LTZ().notNull()),
+                                        DataTypes.FIELD("f15", DataTypes.DECIMAL(10, 2).notNull()),
+                                        DataTypes.FIELD(
+                                                "f16",
+                                                DataTypes.MAP(DataTypes.INT(), DataTypes.STRING())
+                                                        .notNull()),
+                                        DataTypes.FIELD(
+                                                "f17", DataTypes.ARRAY(DataTypes.INT()).notNull()),
+                                        DataTypes.FIELD(
+                                                "f18", DataTypes.ARRAY(DataTypes.INT()).notNull()),
+                                        DataTypes.FIELD(
+                                                "f19",
+                                                DataTypes.MULTISET(DataTypes.INT()).notNull())))
+                        .convertedTo(
+                                Row.class,
+                                Row.ofKind(
+                                        RowKind.DELETE,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        null,
+                                        new Integer[] {null, null},
```

Comment:
Shouldn't this be null instead?

Reply:
I test both `null` and an array with `null` values

Reply:
I should update the element type to be not null, though